src/EventStore.Core/Services/Transport/Grpc/Enumerators.AllSubscription.cs

```diff
@@ -217,13 +215,10 @@ public class AllSubscription : IAsyncEnumerator<ReadResp> {
 				Task.Factory.StartNew(PumpLiveMessages, _cancellationToken);
 
 				async Task PumpLiveMessages() {
-					var position = await caughtUpSource.Task.ConfigureAwait(false);
+					await caughtUpSource.Task.ConfigureAwait(false);
 
 					await _channel.Writer.WriteAsync(new ReadResp {
-						Checkpoint = new ReadResp.Types.Checkpoint {
-							CommitPosition = position.CommitPosition,
-							PreparePosition = position.PreparePosition
-						}
+						CaughtUp = new ReadResp.Types.CaughtUp()
```

Comment:
Do we definitely want to send the CaughtUp instead of the Checkpoint? it might be better to send both because the checkpoint was presumably there for a reason, but I am not 100% certain

Reply:
So, I had a conversation with @thefringeninja.
`i forget who mentioned it but there is a bug there where we send an extra checkpoint once the user has caught up live. that behavior should be removed`
`yes because if you subscribe from that checkpoint you will miss something. so right now it's at most once delivery which is extremely wrong`

Reply:
ahh right, thanks!